Improvements in @filterable and @sortable docs (#193)
graphile/graphile.github.io: src/pages/postgraphile/smart-tags.md

Applies to:

> - Functions returning `SETOF`: adds the `orderBy` argument to this connection,

Reviewer: ```suggestion
- Functions returning `SETOF` a composite type (e.g. a table type, record type, or another type with "column"-like attributes): adds the `orderBy` argument to this connection,
```

>   allowing to _sort_ the set by any of its scalar fields.

Reviewer: ```suggestion
  allowing to _sort_ the set by its columns/attributes.
```

I'd rather not say "any" here, as things like `ignoreIndexes: false` may remove certain ones.

- Computed column functions with no required parameters and returning a scalar
  or array: allows this function to be used as an order in the `orderBy`

>   argument for the parent table, allowing to _sort by_ this function's value the

Reviewer: ```suggestion
  argument for the parent table, allowing to _sort_ the parent table _by_ this function's return value.
```

  parent table.

If used on a computed function that returns a SETOF _record type_ it will make
the connection _sortable_ (like it would with any other `SETOF` function) and

> NOT a _sort by_ for its parent table

Reviewer: With the edits above, I don't think this paragraph is necessary? It might be helpful to say that `@sortable has no effect on functions that return SETOF scalar` or similar?

Author: I agree, not necessary anymore. I'll add a warning on scalar `SETOF`, better to cover as much as possible to minimise questions

comment on function user_object_field() is E'@sortable';
```

Note that this will only work with a single composite type. If your computed

> function returns a `SETOF` composite type, you will need to use an `ARRAY` and

Reviewer: ```suggestion
function returns a `SETOF` composite type one option is to use an `ARRAY` and
```

the
[connection-filter plugin](https://github.com/graphile-contrib/postgraphile-plugin-connection-filter).

```sql
-- non scalar setof function
CREATE OR REPLACE FUNCTION user_objects(user user) RETURNS SETOF object AS $$
SELECT * FROM object where owner_id = user.id; -- one user, many objects
$$ language SQL STABLE;

-- wrapper. Note the () for notation. Failing to use them will throw an error
CREATE OR REPLACE FUNCTION user_object_fields(user user) RETURNS varchar[] AS $$
SELECT ARRAY_AGG((user_objects(user)).field);
$$ language SQL STABLE;

-- don't forget the comments...
comment on function user_object_fields() is E'@sortable';
```

Now you can use the array as a _filter by_, for example:

```gql
{
  # get all users who own an object with field == 'foo'
  allUsers(filter: {object_fields: contains: 'foo'}) { ... }
}

> ```

Reviewer: This section belongs in the postgraphile-plugin-connection-filter project; I typically don't recommend the addition of arbitrary filters to a GraphQL API, so I'd rather not have this example so prominently in our docs. Happy to link out to it though :+1:

Author: Would you be ok to leave the example with a big warning saying something like "this is not recommend, but in case you want to do it..."? 

As a user, I prefer to see how things are done and understand the cost of doing it than to hide it, but I also understand your opinion on not showing it.

Collateral question, what do you recommend? A custom query with the filter parameter? We could also show an example on the recommended way to do it

Reviewer: Yeah I'd probably use a custom query, or possibly https://www.graphile.org/postgraphile/make-add-pg-table-condition-plugin/
